specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,318 @@
+                                info.lpFile = protocol_url.c_str();
+                                info.nShow = SW_SHOWNORMAL;
+                                ::ShellExecuteEx(&info);
+                            }
+                        }
+                        return S_OK;
+                    };
+                    wil::com_ptr<ICoreWebView2Deferral> deferral;
+                    CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                    m_appWindow->RunAsync(
+                        [deferral, showDialog]()
+                        {
+                            showDialog();
+                            CHECK_FAILURE(deferral->Complete());
+                        });
+                    return S_OK;
+                })
+                .Get(),
+            &m_launchingExternalProtocolToken));
+    }
+}
+    
+```
+
+## .NET and WinRT
+
+```c#
+private WebView2 webView;
+void RegisterLaunchingExternalProtocolHandler() 
+{
+    webView.CoreWebView2.LaunchingExternalProtocol += (sender, args) {
+        {
+            CoreWebView2Deferral deferral = args.GetDeferral();
+            System.Threading.SynchronizationContext.Current.Post((_) =>
+            {
+                using (deferral)
+                {
+                    // Set the `Cancel` property to `TRUE`, as we will either silently launch the 
+                    // trusted app or display a custom dialog. 
+                    args.Cancel = true;
+                    if (String.Equals(args.Uri, "calculator:///", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // If this matches our desired protocol, then set the
+                        // event args to cancel the event and launch the
+                        // calculator app.
+                        ProcessStartInfo info = new ProcessStartInfo
+                        {
+                            FileName = args.Uri,
+                            UseShellExecute = true
+                        };
+                        Process.Start(info);
+                    } 
+                    else
+                    {
+                        // To display a custom dialog we cancel the launch, display
+                        // a custom dialog, and then manually launch the external protocol. 
+                        string text = "Launching External Protocol";
+                        if (args.InitiatingOrigin != "")
+                        {
+                            text += "from ";
+                            text += args.InitiatingOrigin;
+                        }
+                        text += " to ";
+                        text += args.Uri;
+                        text += "\n";
+                        text += "Do you want to grant permission?";
+                        string caption = "Launching External Protocol request";
+                        MessageBoxButton btnMessageBox = MessageBoxButton.YesNoCancel;

```

Comment:
```suggestion
                        MessageBoxButton btnMessageBox = MessageBoxButton.YesNo;

```

No point having a Cancel button that means the same as No.

Reply:
Please fix.